crates/sui/src/sui_commands.rs

```diff
@@ -203,6 +203,17 @@ pub enum SuiCommand {
         #[clap(long)]
         epoch_duration_ms: Option<u64>,
 
+        /// Make the fullnode dump executed checkpoints as files to a local directory. This is
+        /// incompatible with --no-full-node.
+        #[clap(
+            long,
+            default_missing_value_os = tempdir().unwrap().path().as_os_str().to_owned(),
```

Comment:
I think this might create the tempdir even if we don't need it? Better to allow this to be `None` and we can create it in the body of `start` if necessary.

Reply:
I had before as 
```rust
        #[clap(
            long,
            num_args = 0..=1,
            require_equals = true,
            value_name = "DATA_INGESTION_DIR",
        )]
        with_data_ingestion: Option<Option<PathBuf>>,
```
Is that what you had in mind?

Reply:
I'll try the above. If needed, I can revert the commit before merging

Reply:
I was expecting you would only need the one level of `Option<...>`, but apart from that, that's what I had in mind!

Reply:
No worries. With stefan's suggestion, this will be much like that